### PR TITLE
Predefine file extensions to gzip. Make the list customizable.

### PR DESCRIPTION
**Pull request by koresar**
* GZIp all the files if `--gzip` is provided. If file extension list is given then gzip only those.
* Add corresponding unit tests.
* Update README accordingly.

Should be published as v1.1.

### CODE REVIEW

File: src/deploy.js [hunk 1]
```diff
@@ -153,7 +166,7 @@ export const readFile = co.wrap(function *(filepath, cwd, gzipFiles) {
  * and uploading files that are not there yet, or do need an update.
  */
 export const handleFile = co.wrap(function *(filePath, cwd, filePrefix, client, s3Options, ext, indexName, preventUpdates) {
-  const fileObject = yield readFile(filePath, cwd, s3Options.ContentEncoding !== undefined);
+  const fileObject = yield readFile(filePath, cwd, s3Options.gzip);
```
Comment (domderen) [inline src/deploy.js]: If I trace the code correctly, `s3Options` does not have the `gzip` attribute. That attribute exists on the `options` object in the `deploy` function, but it is never passed to `s3Options`. Can you verify?

Comment (koresar) [reply src/deploy.js]: Oh, mate! I'm so sorry! There are 5 "options" objects I got confused. And all the babel compilation thing does not allow me to test things.
`options, AWSOptions, s3Options, s3ClientOptions, cfOptions`

Let me come up with a better unit test for that part.

Sorry again.

**Comment by koresar** [comment]
Hey @domderen 

Please, take a look at this commit: https://github.com/import-io/s3-deploy/pull/57/commits/48d7680e22e4af566dd1105e48735a79e66eb4c6

What I did:
1) Took out `AWSOptions, s3Options, s3ClientOptions, cfOptions` object creation from `index.js`
2) Copied them to `deploy.js`
3) Pass only one variable - `options` - to the `deploy(option)` function.

Nothing else was changed.

(I copied across the the PR you merged 10 hours ago.)

**Comment by domderen** [comment]
Thanks for that, I'll take a closer look soon ;)

**Comment by koresar** [comment]
@domderen proudly present you The Unit Test for the uploading process! :)
https://github.com/import-io/s3-deploy/pull/57/files#diff-d24ad2ebc2d9b84b5a2e019d6ad421ce

It's a bit hacky, but the mixer of FP and complex I/O is hard to unit test in general.

**Comment by domderen** [comment]
Thanks very much for your contribution!

**Pull request merged**